persist/fs/retriever.go

```diff
@@ -423,12 +427,17 @@ func (r *blockRetriever) shardRequests(
 
 func (r *blockRetriever) Close() error {
 	r.Lock()
-	defer r.Unlock()
-
 	r.nsMetadata = nil
 	r.status = blockRetrieverClosed
```

Comment:
Perhaps test if `if r.status == blockRetrieverClosed { return errBlockRetrieverAlreadyClosed }`?  Especially now that `close(...)` will cause a panic now if this is called twice.

Reply:
good idea